src/main/java/nus/climods/MainApp.java

```diff
@@ -76,12 +77,30 @@ public void init() throws Exception {
      * data from the sample address book will be used instead if {@code storage}'s address book is not found, or an
      * empty address book will be used instead if errors occur when reading {@code storage}'s address book.
      */
-    private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
+    private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs)
+            throws IOException {
         Optional<ReadOnlyModuleList> moduleListOptional;
         ReadOnlyModuleList initialModuleList;
-        // TODO: load from storage
-        UniqueUserModuleList initialUserModuleList = new UniqueUserModuleList();
+        Optional<UniqueUserModuleList> userModuleListOptional;
+        UniqueUserModuleList initialUserModuleList;
         String academicYear = userPrefs.getAcademicYear();
+
```

Comment:
This seems to be duplicate logic here for loading `UserModule` and `ModuleList`, maybe if you have the time we can try abstracting this out and have a helper instead?

Reply:
I should be able to do something about this maybe tmr? Let me make a commit for this

Reply:
good catch! I was too tired to fix it ystd, I have made a commit to abstract the loading of stored lists